File: pset1/mario/less/mario.c
// ------------------------------------------ //
// CS50 Problem Set 1: Mario (less)
//
// Summary: Prints Mario`s pyramid (less)
//
// Lucas Emidio Fernandes Dias
// 9 March 2018
// ------------------------------------------ //


// Include libraries
#include <stdio.h>
#include <cs50.h>

// ------------------------------------------ //
//        MAIN                                //
// ------------------------------------------ //
int main(void)
{
    int height; // Height of the pyramid (user input)

    // Asks the user for a height
    do
    {
        height = get_int("Height: ");
    }
    // Height is only valid if 0 <= height <= 23
    while (height < 0 || height > 23);

    // Prints the pyramid
    for (int i = 0; i < height; i++)
    {
        // Print spaces
        for (int j = 0; j < height - i - 1; j++)
        {
            printf(" ");
        }
        // Print hashes
        for (int j = 0; j < i + 2; j++)
        {
            printf("#");
        }
        // Print new line
        printf("\n");
    }
}

File: pset6/mario/less/mario.py
# ----------------------------------------------
# CS50 Problem Set 6: Sentimental - Mario (less)
#
# Summary: Reimplements Masrio (less) in python
# Prints Mario`s pyramid (less)
#
# Lucas Emidio Fernandes Dias
# 03 October 2018
# ----------------------------------------------

from cs50 import get_int

# Asks the user for a height until height is valid
while True:

    height = get_int("Height: ")

    # Height is only valid if 0 <= height <= 23
    if height >= 0 and height <= 23:
        break

# Prints the pyramid
for i in range(height):

    # Print spaces
    print(" " * (height - 1 - i), end="")

    # Print hashes
    print("#" * (i + 2), end="")

    # Print new line
    print()
